### ISSUE TITLE
Standalone 'Customizer'-style app

### ISSUE THREAD

**Issue by DeadlyDad** [codelv/declaracad#49]
One of the great new features of OpenSCAD is the 'Customizer', which allows quickly altering models based on adjustable parameters.  It occurred to me that a **fantastic** option for DeclaraCAD would be an 'export standalone model' option, which (perhaps using [Auto PY to EXE](https://github.com/brentvollebregt/auto-py-to-exe)) would create an app that behaved similarly to OpenSCAD's Customizer, and included the ability to export the customized models in the various formats.  This would allow users to alter models without needing to go through the hassle of installing DeclaraCAD on their systems, as well as giving it a **LOT** of exposure.  

(The next obvious platforms would be to create web-and Blender-based viewer/exporters.)

Does that make sense?

![image](https://user-images.githubusercontent.com/13209333/102532664-29a6db00-4059-11eb-8a82-19a5ed8ae5c6.png)

**Comment by frmdstryr** [comment]
You could make a separate app that just uses declaracad as a library and embeds the declaracad viewer.  This is how the DeclaraCAD UI works.

A web backend would be nice but I'm pretty sure that requires either recompiling OCCT and pyOCCT to use egl or to do it how python-occ does it (last I looked I think they were serializing the models and rendering them with three.js).

This sort of goes against the point of DeclaraCAD though... which is to write models in code. If people want to use a UI there's a lot better options (like OnShape, Fusion, etc..)

**Comment by DeadlyDad** [comment]
Hmmm...  I guess I didn't communicate properly.  I'm not suggesting an editor, but a viewer/exporter; not--to use an analogy--Acrobat itself, but Acrobat *Reader*, something that non-modellers can use to view, customize, and export models that someone else created.  In the example image, end-users can change a few basic parameters that alter the model within set boundaries **without** needing to know how to script.  Does that make more sense?

**Comment by frmdstryr** [comment]
Yes, I see.  I like the idea.  If the goal is to make it easier for non-technical users to easily view, customize, and export predefined models would a web based viewer be the most ideal?  

The customizer app would still need to have all the same dependencies so the app wouldn't be any smaller. Instead of a separate app maybe there should just be a separate "viewer" workbench.
